### PR TITLE
ACR Updates from Arch Board review

### PR DESCRIPTION
**Pull request by annelo-msft**
Contributes to https://github.com/Azure/azure-sdk-for-net/issues/34958

This PR includes the following subset of changes recommended by architects:
- Move ContainerRegistryBlobClient into main namespace
- Change `repository` parameter in constructor to `repositoryName` to match main client
- Change parameter names to blob methods to `content` and to manifest methods to `manifest`
- Change URI properties on OciAnnotations to `Uri` type
- Change `Config` to `Configuration`
- Update doc comments to indicate which response is returned when service method makes multiple calls

### CODE REVIEW

**Review by lmolkova — COMMENTED** [review]
I should have raised it during arch-board, but I'd prefer to keep `config` and not rename it to `configuration` - this is how it's called in the [oci spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md). Divergence from the spec terminology seems to hurt user experience more than improve it.

Other that that changes look great to me!

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/DownloadManifestResult.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 using System.IO;
 using System.Text.Json;
 
-namespace Azure.Containers.ContainerRegistry.Specialized
+namespace Azure.Containers.ContainerRegistry
 {
     /// <summary>
     /// The result from downloading an OCI manifest from the registry.
```
Comment (lmolkova) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/DownloadManifestResult.cs:11]: nit and out of scope on this PR: wonder if we should rename `AsOciManifest` to `AsOciImageManifest` across the languages (so we can add `AsOciArtifactManifest` if/when artifact manifest comes)

Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/DownloadManifestResult.cs:11]: The feedback from Ted was that we should change the name of this method to make it clear it is deserializing the BinaryData Content property.

**Review by lmolkova — APPROVED** [review]
LGTM per offline discussion!

**Pull request merged**